app/src/ble.c

```diff
@@ -183,6 +185,12 @@ int update_advertising() {
     struct bt_conn *conn;
     enum advertising_type desired_adv = ZMK_ADV_NONE;
 
+    uint8_t max_device_name_length = (active_profile > 9) ? 13 : 14;
+    // need to store the temporary device name because Zephyr *printf doesn't support string precision
+    char temp_device_name[14];
+    snprintf(temp_device_name, max_device_name_length, "%s", DEVICE_NAME);
+    snprintf(device_name, 16, "%s~%d", temp_device_name, active_profile);
```

Comment:
We can probably avoid the double copy here with something like:

```
uint8_t prof_num_len = active_profile > 9 : 3 : 2;
snprintf(device_name, MAX_NAME - prof_num_le, "%s", DEVICE_NAME);
sprintf(device_name + MAX_NAME - prof_num_len, prof_number_len, "-%d", active_profile);
```

Just specifying a different start address for the snprintf for the second call, to just print it into the "end" of the string.